state/factory/statedb.go

```diff
@@ -469,17 +495,27 @@ func (sdb *stateDB) States(opts ...protocol.StateOption) (uint64, state.Iterator
 	if cfg.Key != nil {
 		return sdb.currentChainHeight, nil, errors.Wrap(ErrNotSupported, "Read states with key option has not been implemented yet")
 	}
-	if cfg.Cond == nil {
-		cfg.Cond = func(k, v []byte) bool {
-			return true
+	var values [][]byte
+	if cfg.Keys == nil {
```

Comment:
add a comment like
// no Keys set, default to get all key-values in the namespace